proposals/0069_prometheus-exporters-otel-embedding.md

```diff
@@ -0,0 +1,286 @@
+
+### OpenTelemetry Collector Receiver Implementation
+
+The `exporter-toolkit` will provide a complete implementation of OpenTelemetry's receiver interfaces:
+
+1. **component.Factory** - for component type and default configuration
+2. **component.Component** - for lifecycle management
+3. **receiver.Factory** - for creating receiver instances
+4. **receiver.Metrics** - for producing pmetric data
+
+This implementation will:
+- Start the Prometheus exporter and obtain its Registry
+- Run a periodic scrape loop based on the configured interval
+- Convert scraped metrics to pmetric format
+- Push metrics to the OpenTelemetry pipeline consumer
+
+### Using with OpenTelemetry Collector Builder
+
+Once a Prometheus exporter implements the new interfaces, it can be included in custom OpenTelemetry Collector distributions via OCB:
+
+```yaml
+# ocb-config.yaml
+receivers:
+  - gomod: github.com/prometheus/node_exporter v1.x.x
+```
+
+The OCB will recognize the exporter as a valid receiver and include it in the built collector binary.
+
+### Example Configuration
+
+In the OpenTelemetry Collector configuration:
+
+```yaml
+receivers:
+  node_exporter:
+    scrape_interval: 30s
+    exporter_config:
+      # Node exporter specific configuration
+      collectors:
+        - cpu
+        - diskstats
+        - filesystem
+
+exporters:
+  otlp:
+    endpoint: otelcol:4317
+
+service:
+  pipelines:
+    metrics:
+      receivers: [node_exporter]
+      exporters: [otlp]
+```
+
+### Implementation Steps
+
+1. **Extend exporter-toolkit** with the new interfaces and OpenTelemetry Collector receiver implementation
+2. **Implement the Prometheus to pmetric converter** (potentially adapting existing code)
+3. **Update one reference exporter** (e.g., node_exporter) to implement the new interfaces as a proof of concept
+4. **Validate** with OpenTelemetry Collector Builder
+5. **Document** the integration pattern for other exporters
+6. **Gradually adopt** across other Prometheus exporters based on community interest
+
+### Migration and Compatibility
+
+* **No breaking changes** to existing Prometheus exporters that don't adopt the interfaces
+* **Opt-in adoption** - exporters can choose when/if to implement embedding support
+* **Backward compatibility** - embedded exporters still work as standalone exporters
+
+
+### Known Problems
+
+1. **Dependency conflicts**: Prometheus exporters and OpenTelemetry collector-contrib use different dependency versions. Building a distribution with both may require dependency alignment or replace directives.
+
+2. **Scope of adoption**: It's unclear how many Prometheus exporters will adopt these interfaces. The proposal targets exporters in the `prometheus` and `prometheus-community` GitHub organizations initially.
+
+3. **Metric semantics**: Subtle differences in how Prometheus and OpenTelemetry handle certain metric types may require careful mapping.
```

Comment:
I see a source of friction in the different naming conventions.

The idea with naming schemas could complement this nicely and should maybe mentioned here.